parking_permits/views.py

```diff
@@ -162,6 +163,85 @@ def post(self, request, format=None):
         return Response(res)
 
 
+class TalpaResolveProduct(APIView):
+    @swagger_auto_schema(
+        operation_description="Resolve product for subscription.",
+        request_body=TalpaPayloadSerializer,
+        responses={
+            200: openapi.Response("Resolve product", ResolveProductResponseSerializer)
+        },
+        tags=["ResolveProduct"],
+    )
+    def post(self, request, format=None):
+        logger.info(
+            f"Data received for resolve product = {json.dumps(request.data, default=str)}"
+        )
+        subscription_id = request.data.get("subscriptionId")
+        if not subscription_id:
+            return bad_request_response("Subscription id is missing from request data")
+        user_id = request.data.get("userId")
+        if not user_id:
+            return bad_request_response("User id is missing from request data")
+        order_item_data = request.data.get("orderItem")
+        if not order_item_data:
+            return bad_request_response("Order item data is missing from request data")
+        order_item_id = order_item_data.get("orderItemId")
+        if not order_item_id:
+            return bad_request_response("Order item id is missing from request data")
+        meta = order_item_data.get("meta")
```

Comment:
Is `meta` used anywhere in this function? Why is it needed in the POST?

Reply:
@danjacob-anders Permit id is fetched from inside the `meta` in line:
`permit_id = get_meta_value(meta, "permitId")`
So thus we need it.